nntrainer/src/neuralnet.cpp

```diff
@@ -132,71 +117,23 @@ int NeuralNetwork::setConfig(std::string config) {
   return status;
 }
 
-int NeuralNetwork::loadFromConfig() {
-  int status = ML_ERROR_NONE;
-  std::string ini_file = config;
-  int num_ini_sec = 0;
+int NeuralNetwork::loadNetworkConfig(void *_ini) {
+  dictionary *ini = static_cast<dictionary *>(_ini);
   char unknown[] = "Unknown";
-  char model_name[] = "model.bin";
-  dictionary *ini;
-  std::vector<std::string> section_names;
-  std::vector<std::string>::iterator section_names_iter;
-
-  if (ini_file.empty()) {
-    ml_loge("Error: Configuration File is not defined");
-    return ML_ERROR_INVALID_PARAMETER;
-  }
-
-  /** Parse ini file */
-  ini = iniparser_load(ini_file.c_str());
-  if (ini == NULL) {
-    ml_loge("Error: cannot parse file: %s\n", ini_file.c_str());
-    return ML_ERROR_INVALID_PARAMETER;
-  }
-
-  /** Get number of sections in the file */
-  num_ini_sec = iniparser_getnsec(ini);
-  if (num_ini_sec < 0) {
-    ml_loge("Error: invalid number of sections.");
-    return ML_ERROR_INVALID_PARAMETER;
-  }
-
-  /** Get all the section names */
-  for (int idx = 0; idx < num_ini_sec; ++idx) {
-    const char *sec_name = iniparser_getsecname(ini, idx);
-    if (!sec_name) {
-      ml_loge("Error: Unable to retrieve section names from ini.");
-      return ML_ERROR_INVALID_PARAMETER;
-    }
-    std::string sec_name_lower(sec_name);
-    std::transform(sec_name_lower.begin(), sec_name_lower.end(),
-                   sec_name_lower.begin(),
-                   [](unsigned char c) { return std::tolower(c); });
-    section_names.push_back(sec_name_lower);
-  }
-
-  /** Parse the Network section and its properties */
-  section_names_iter =
-    std::find(section_names.begin(), section_names.end(), "network");
```

Comment:
this is necessary to check that "network" section exists in the ini file.
otherwise we will have a default network created rather than giving an error.